File: 382/Python/Main.py
import random

class Solution(object):

    def __init__(self, head):
        self.head = head
        

    def getRandom(self):
        count = 0
        currenthead = self.head
        answer = 0
        while currenthead:
            if random.randint(0,count) == 0:
                answer = currenthead.val
            currenthead = currenthead.next
            count += 1
        return answer
        

File: 382/C++/main.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */

#include <stdlib.h> 

class Solution {
public:
    /** @param head The linked list's head.
        Note that the head is guaranteed to be not null, so it contains at least one node. */
    Solution(ListNode* head) {
        curhead = head;
    }
    
    /** Returns a random node's value. */
    int getRandom() {
        int count = 1;
        int random;
        ListNode *Current = curhead;
        while (Current) {
            if((rand() % count) == 0) {
                random = Current->val;
            }
            Current = Current->next;
            count++;
        }
        return random;
    }
private:
ListNode *curhead;  
};
